exploring_recommendation_system_algorithms/article.md

```diff
@@ -0,0 +1,114 @@
+
+## **Exploring Recommendation System Algorithms: From Classic to Cutting-Edge**
+
+![](https://cdn-images-1.medium.com/max/2000/1*U84sFtkay34IqI_dOAkWDw.jpeg)
+
+## Introduction
+
+  In the past, people generally purchased products recommended to them by their friends or the people they trust. This is how people used to make purchasing decisions when there was doubt about a product. But since the advent of the Internet, we are so used to ordering online and streaming music and movies that we are constantly creating data in the back end. A recommendation engine uses that data and different algorithms to recommend the most relevant items to users. It initially captures the past behavior of a user, and then it recommends items for future purchase or use.
+
+There are many different types of recommendation engines, the following are the most commonly used and widely recognized in various industries, and each of them is explored in this article.
+
+*  **Content-based filtering**
+
+*  **Collaborative-based filtering**
+
+* **Deep learning recommendation system**
+
+* **Graph Based recommendation system**
+
+## Content-Based Filtering
+
+ The content-based filtering method is a recommendation algorithm that suggests items similar to the ones the users have previously selected or shown interest in. It can recommend based on the actual content present in the item. For example, as shown in Figure 2, a new article is recommended based on the text present in the articles
+
+![Figure 2, Content-Based Filtering](https://cdn-images-1.medium.com/max/2000/1*a4hScSGe2G762WtX78i0xg.png)
+
+Let’s take a peek at how Netflix suggests what you might want to watch. Imagine they have two special lists: one for what you like (profile) and one for all their shows (item). To match these lists, they use a simple trick called “cosine similarity.”, formula is shown in Figure 2.2. It’s like measuring how close two things are. They check how similar your likes are to the details of the shows. If your likes and a show’s details are similar, Netflix suggests it. This helps them recommend stuff you’re likely to enjoy, based on what you’ve liked before.
+
+![Figure 2.2 Cosine Similarity formula](https://cdn-images-1.medium.com/max/2000/1*MMb3QqZCFpszitGcBOmMEQ.png)

```

Comment:
I would recommend to write the formula in LaTeX formatting rather than image in this kind of cases